core/src/main/java/software/amazon/awssdk/async/StringAsyncResponseHandler.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2010-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.async;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import org.reactivestreams.Publisher;
+import software.amazon.awssdk.annotation.SdkInternalApi;
+
+/**
+ * Implementation of {@link AsyncResponseHandler} that dumps content into a UTF8 encoded string.
+ *
+ * @param <ResponseT> Pojo response type.
+ */
+@SdkInternalApi
+class StringAsyncResponseHandler<ResponseT> implements AsyncResponseHandler<ResponseT, String> {
+
+    private final AsyncResponseHandler<ResponseT, byte[]> byteArrayResponseHandler;
+
+    StringAsyncResponseHandler(
+            AsyncResponseHandler<ResponseT, byte[]> byteArrayResponseHandler) {
+        this.byteArrayResponseHandler = byteArrayResponseHandler;
+    }
+
+    @Override
+    public void responseReceived(ResponseT response) {
+        byteArrayResponseHandler.responseReceived(response);
+    }
+
+    @Override
+    public void onStream(Publisher<ByteBuffer> publisher) {
+        byteArrayResponseHandler.onStream(publisher);
+    }
+
+    @Override
+    public void exceptionOccurred(Throwable throwable) {
+        byteArrayResponseHandler.exceptionOccurred(throwable);
+    }
+
+    @Override
+    public String complete() {
+        return new String(byteArrayResponseHandler.complete(), StandardCharsets.UTF_8);
```

Comment:
Because this is their data, we should probably make them define the encoding.

Reply:
+1, I'll add that.